nucliadb_core/src/query_planner.rs

```diff
@@ -175,8 +167,18 @@ pub fn build_query_plan(search_request: SearchRequest) -> NodeResult<QueryPlan>
     let vectors_context = VectorsContext {
         filtering_formula: search_query.clone(),
     };
+
+    let filtering_formula = match (search_query, prefilter_query.clone()) {
+        (None, None) => None,
+        (Some(formula), None) | (None, Some(formula)) => Some(formula),
+        (Some(search), Some(prefilter)) => Some(BooleanExpression::Operation(BooleanOperation {
+            operator: query_language::Operator::And,
+            operands: vec![search, prefilter],
+        })),
+    };
+
     let paragraphs_context = ParagraphsContext {
-        filtering_formula: search_query,
+        filtering_formula,
     };
```

Comment:
This can only be done for paragraphs version 2, version 1 would need to use the prefiltered keys since it.